pkg/engine/transform/transform.go

```diff
@@ -539,6 +541,26 @@ func (t *Transformer) NormalizeResourcesForK8sMasterUpgrade(logger *logrus.Entry
 					filteredResources = filteredResources[:len(filteredResources)-1]
 				}
 			}
+		} else if resourceType == roleResourceType {
+			logger.Infoln(fmt.Sprintf("Evaluating if agent resource: %s needs to be removed", resourceName))
+			removeRole := true
+			// The usage of NormalizeResourcesForK8sMasterUpgrade across the code base seems to indicate that
+			// agentPoolsToPreserve == nil => master node upgrade
```

Comment:
do you mean if agentPoolsToPreserve has a length of 0?

Reply:
you are right, I wrote it with this in mind

https://github.com/Azure/aks-engine/blob/v0.57.0/pkg/operations/kubernetesupgrade/upgrader.go#L201